modules/google-speech/src/backend/middleware.ts

```diff
@@ -0,0 +1,139 @@
+    // TODO: Add more validation than the payload type being audio
+    if (event.payload.type !== 'audio') {
+      return next()
+    }
+
+    const client: GoogleSpeechClient = this.clients[event.botId]
+    if (!client) {
+      return next()
+    }
+
+    const audioFile = event.payload.audio
+    if (audioFile) {
+      try {
+        // TODO: Fetch bot current language too?
+        const language: string = event.state.user.language?.replace(/'/g, '')
+        const text: string = await client.speechToText(audioFile, language)
+
+        if (text) {
+          const newEvent: sdk.IO.Event = this.bp.IO.Event({
+            type: event.type,
+            direction: event.direction,
+            channel: event.channel,
+            target: event.target,
+            threadId: event.threadId,
+            botId: event.botId,
+            payload: { type: 'text', text, textToSpeech: true }
+          })
+
+          await this.bp.events.sendEvent(newEvent)
+
+          return next(null, true)
+        }
+      } catch (err) {
+        this.bp.logger.forBot(event.botId).error('[speech-to-text]:', err)
+        return next(err)
+      }
+    }
+
+    return next()
+  }
+
+  private async outgoingHandler(event: sdk.IO.OutgoingEvent, next: sdk.IO.MiddlewareNextCallback) {
+    const incomingEventId: string = event.incomingEventId
+    if (!incomingEventId || event.type !== 'text') {
+      return next()
+    }
+
+    const incomingEvents: sdk.IO.StoredEvent[] = await this.bp.events.findEvents({
+      incomingEventId: event.incomingEventId,
+      direction: 'incoming'
+    })
+    if (!incomingEvents.length || incomingEvents[0].event.payload.textToSpeech !== true) {
+      return next()
+    }
+
+    const client: GoogleSpeechClient = this.clients[event.botId]
+    if (!client) {
+      return next()
+    }
+
+    const text = event.payload.text
+    if (text) {
+      try {
+        // TODO: Fetch bot current language too?
+        const userAttributes = await this.bp.users.getAttributes(event.channel, event.target)
+        const language: string = userAttributes['language']?.replace(/'/g, '')
+        const audio = await client.textToSpeech(text, language)
+
+        const data = new FormData()
+        data.append('file', audio, `${uuidv4()}.mp3`)
+
+        // TODO: Add cache (preview -> buffer)
+        const axiosConfig = await this.bp.http.getAxiosConfigForBot(event.botId, { localUrl: true })
+        axiosConfig.headers['Content-Type'] = `multipart/form-data; boundary=${data.getBoundary()}`
+
+        const res = await axios.post<{ url: string }>('/media', data, {
+          ...axiosConfig
+        })
+
+        const newEvent: sdk.IO.Event = this.bp.IO.Event({
+          type: 'audio',
+          direction: event.direction,
+          channel: event.channel,
+          target: event.target,
+          threadId: event.threadId,
+          botId: event.botId,
```

Comment:
you can replace those 5 lines with `..._.pick(event, ['direction', 'channel', 'target', ...]`